askup/views.py

```diff
@@ -1,24 +1,53 @@
 """Askup django views."""
 import logging
 
-from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth import logout
 from django.contrib.auth.decorators import login_required
 from django.contrib.auth.models import User
+from django.http import Http404, JsonResponse
 from django.shortcuts import get_object_or_404, redirect, render
 from django.urls import reverse
-from django.utils.decorators import method_decorator
 from django.views import generic
 
-from .forms import OrganizationModelForm, QsetDeleteModelForm, QsetModelForm, UserLoginForm
-from .mixins import ListViewUserContextDataMixin, QsetViewMixin
-from .models import Organization, Qset, Question
-from .utils import user_group_required
+from .forms import (
+    AnswerModelForm,
+    OrganizationModelForm,
+    QsetDeleteModelForm,
+    QsetModelForm,
+    QuestionDeleteModelForm,
+    QuestionModelForm,
+    UserLoginForm,
+)
+from .mixins.views import (
+    CheckSelfForRedirectMixIn,
+    ListViewUserContextDataMixIn,
+    QsetViewMixIn,
+    UserFilterMixIn,
+)
+from .models import Answer, Organization, Qset, Question
+from .utils.general import (
+    add_notification_to_url,
+    check_user_has_groups,
+    get_user_answers_count,
+    get_user_questions_count,
+    get_user_score_by_id,
+)
+from .utils.views import (
+    compose_qset_form,
+    compose_question_form_and_create,
+    delete_qset_by_form,
+    qset_update_form_template,
+    question_vote,
+    user_group_required,
+    validate_and_send_feedback_form,
+    validate_answer_form_and_create,
+)
 
 
 log = logging.getLogger(__name__)
 
 
-class OrganizationsView(generic.ListView):
+class OrganizationsView(CheckSelfForRedirectMixIn, generic.ListView):
```

Comment:
login_required?

Reply:
I did it by the MixIn because this way we can prevent database query execution on user permissions fail.